discovery/discovery_test.go

```diff
@@ -96,18 +98,41 @@ func TestNewDBOrchestratorPoolCache_GivenListOfOrchs_CreatesPoolCacheCorrectly(t
 	require.NotNil(dbOrch)
 }
 
+type stubRpc struct {
+}
+
+func (s *stubRpc) getOrchestrator(ctx context.Context, orchestratorServer *url.URL) (*net.OrchestratorInfo, error) {
+	return &net.OrchestratorInfo{Transcoder: orchestratorServer.String()}, nil
+}
+
+func TestTempGetOrchestrators_GivenOneOrchestratorRequested_ReturnsRandomOrch(t *testing.T) {
```

Comment:
I don't think this test currently actually checks that a random O is returned since it just checks that an O was returned

Reply:
Yes, this is true. I did give this a lot of thought on Friday.

My initial thinking was that I could just call `NewOrchestratorPool` and `GetOrchestrators(1)` multiple times in a row, compare outcomes, and assure that at least one of the resulting `orchInfo[0].Transcoder` addresses from those calls would be different. But I had some concerns:

- Calling those two functions, three times, in the same test, was yielding the same ordering of URIs for me, always. I have since figured out that `r := rand.New(rand.NewSource(time.Now().Unix()))` could be swapped out for `r := rand.New(rand.NewSource(time.Now().UnixNano()))`, and the resulting`randomizedUris` do indeed come back in a different order.
- Remaining concern: Test might fail sometimes, not because the underlying (non-test) code is faulty. The probability of all three `orchInfo[0].Transcoder` addresses being the same (with a sample size of 3 orchAddr) is real. I could increase the number of `orchAddr` I include here. Is this a valid concern? Any other ideas?

Reply:
We really want tests to be deterministic, hence my concerns earlier in the PR about resetting the RNG state. When we control the seed, that gives us better odds of producing deterministic outputs (as long as other code using the same RNG state is doing so in a predictable manner).

As long as output is consistently in an order somewhat different from the original input then I think it's okay. Yes, this approach is somewhat reminiscent of https://xkcd.com/221/  😄

Reply:
I see what you mean now. Didn't really understand the importance of `rand.Perm` before. I'm pushing that fix in a couple of min!